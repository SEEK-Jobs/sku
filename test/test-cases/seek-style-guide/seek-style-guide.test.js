/* eslint-disable jest/expect-expect */
const path = require('path');
const { promisify } = require('util');
const rmfr = require('rmfr');
const fs = require('fs-extra');
const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const waitForUrls = require('../../utils/waitForUrls');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');
const { cwd } = require('../../../lib/cwd');

function createPackageLink(name) {
  return fs.symlink(
    `${cwd()}/node_modules/${name}`,
    `${__dirname}/app/node_modules/${name}`,
  );
}

async function linkLocalDependencies() {
  const nodeModules = `${__dirname}/app/node_modules`;
  await rmfr(nodeModules);
  await fs.mkdir(nodeModules);
  await Promise.all(
    ['react', 'react-dom', 'seek-style-guide'].map(createPackageLink),
  );
}

const assertStorybookContent = async (url) => {
  const page = await browser.newPage();
  await page.goto(url, { waitUntil: 'networkidle2' });

  const content = await page.evaluate(async () => {
    const svg = await window.document
      .querySelector('iframe')
      .contentDocument.querySelector('[data-automation-svg] svg');

    const textElement = await window.document
      .querySelector('iframe')
      .contentDocument.querySelector('[data-automation-text]');
    const text = textElement.innerText;

    return { svg, text };
  });

  expect(content.svg).not.toEqual(null);
  expect(content.text).toEqual('Hello world!');
};

describe('seek-style-guide', () => {
  beforeAll(async () => {
    // "Install" React and seek-style-guide into this test app so that webpack-node-externals
    // treats them correctly.
    await linkLocalDependencies();
    await runSkuScriptInDir('build', appDir);
  });

  it('should generate the expected files', async () => {
    const files = await dirContentsToObject(distDir);
    expect(files).toMatchSnapshot();
  });

  it('should handle seek-style-guide in tests', async () => {
    const { childProcess } = await runSkuScriptInDir('test', appDir);
    expect(childProcess.exitCode).toEqual(0);
  });

  describe('storybook', () => {
    const storybookUrl = 'http://localhost:8081';
    let server;

    beforeAll(async () => {
      server = await runSkuScriptInDir('storybook', appDir, ['--ci']);
      await waitForUrls(storybookUrl);
    });

    afterAll(async () => {
      await server.kill();
    });

    it('should start a storybook server', async () => {
      await assertStorybookContent(storybookUrl);
      expect.assertions(2);
    });
  });
});
