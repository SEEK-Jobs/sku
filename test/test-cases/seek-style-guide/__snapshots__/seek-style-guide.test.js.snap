// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seek-style-guide should generate the expected files 1`] = `
Object {
  "externals.json": "[
  \\"fs\\",
  \\"path\\",
  \\"react\\",
  \\"react-dom/server\\"
]",
  "index.html": SCRIPTS: Array [
    "/runtime-554e53c2f72903d6f45f.js",
    "/vendors~main-7efe22f8be0a6ad2b461.js",
    "/main-3592b1a7a275852ff255.js",
  ]
CSS: Array [
    "/vendors~main-492f2c41d9f41ff813a1.css",
  ]
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="stylesheet"
          href="styles[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <div class="_20YbULa">
        <div class="Oxw3aoN">
          <h1>
            SEEK - Australia&#x27;s no. 1 jobs, employment, career and recruitment site
          </h1>
        </div>
        <div class>
          <div class="_3xxbHuF">
            <div class="GN6nQol">
              <span class="_31viNaz _2R8GxMH _3CnSGKb">
                <span class>
                  <span data-automation-svg="true">
                    <span class="_3q2EWBs">
                      <svg class="URJCV7U"
                           xmlns="http://www.w3.org/2000/svg"
                           width="16"
                           height="16"
                           viewbox="0 0 1024 900"
                           focusable="false"
                      >
                        <path d="M718 188q11 0 22 1.5t22 4.5q16 4 30.5 12t28.5 19q15 13 27 28t21 34h1l3 7q13 27 20 59.5t7 68.5v3q0 15-4.5 32.5T881 496q-11 24-26 47.5T821 590q-39 48-85.5 94T636 774q-46 38-85 67.5T480 891q-31-20-70-49.5T325 774h-1q-52-43-98.5-89T139 590q-19-23-34-46.5T79 496q-10-21-14.5-38.5T60 425v-3q0-37 7-69.5T87 293h1q0-2 1-3.5t2-3.5q10-18 21.5-33t26.5-28q14-11 28.5-19t30.5-12q11-3 22.5-4.5T242 188q24 0 45.5 7t43.5 21q15 9 28 21t24 26q12 15 22.5 32.5T425 333l56 129 54-129q9-20 19-37t22-32q11-14 24.5-26t28.5-22h1q21-14 43-21t45-7zm0-60q-32 0-62 9.5T597 165q-38 25-67.5 61T480 309q-21-47-50-83t-66-61q-29-18-59.5-27.5T242 128q-14 0-29 2t-30 6q-22 6-43 16.5T100 179t-34.5 36T38 259q-1 1-1.5 2t-.5 2l-2 2v1q-17 34-25.5 73T0 422v2q0 22 6 46t19 51q12 27 29 54t39 53q42 52 90.5 100T286 820q49 41 90 71.5t74 51.5q14 9 21.5 13.5t8.5 4.5q1 0 8.5-4.5T510 944q33-22 74-53t90-70q55-46 103.5-94t89.5-99q22-26 39-52.5t29-53.5q13-27 19-51.5t6-46.5v-2q-1-44-9-83t-24-72l-4-8q-13-24-28.5-43.5T860 180t-39.5-27-43.5-17q-15-4-30-6t-29-2z">
                        </path>
                      </svg>
                    </span>
                  </span>
                  <span data-automation-text="true">
                    Hello world!
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
    >
      []
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
    >
    </script>
  </body>
</html>,
  "main-3592b1a7a275852ff255.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime-554e53c2f72903d6f45f.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "vendors~main-492f2c41d9f41ff813a1.css": a,
abbr,
acronym,
address,
applet,
article,
aside,
audio,
b,
big,
blockquote,
body,
canvas,
caption,
center,
cite,
code,
dd,
del,
details,
dfn,
div,
dl,
dt,
em,
embed,
fieldset,
figcaption,
figure,
footer,
form,
h1,
h2,
h3,
h4,
h5,
h6,
header,
hgroup,
html,
i,
iframe,
img,
ins,
kbd,
label,
legend,
li,
mark,
menu,
nav,
object,
ol,
output,
p,
pre,
q,
ruby,
s,
samp,
section,
small,
span,
strike,
strong,
sub,
summary,
sup,
table,
tbody,
td,
tfoot,
th,
thead,
time,
tr,
tt,
u,
ul,
var,
video {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}
article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
  display: block;
}
body {
  line-height: 1;
}
ol,
ul {
  list-style: none;
}
blockquote,
q {
  quotes: none;
}
blockquote:after,
blockquote:before,
q:after,
q:before {
  content: "";
  content: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
mark {
  background-color: transparent;
  color: inherit;
}
html {
  min-height: 100%;
}
._20YbULa,
html {
  background-color: #eee;
}
._20YbULa {
  font-size: 10px;
  font-family: Roboto, Helvetica Neue, HelveticaNeue, Helvetica, Arial,
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #1c1c1c;
  box-sizing: border-box;
}
._20YbULa *,
._20YbULa :after,
._20YbULa :before {
  font-family: inherit;
  box-sizing: inherit;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  -webkit-tap-highlight-color: transparent;
}
._20YbULa input[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
._1hwCpFM {
  height: 100%;
}
.Oxw3aoN {
  position: absolute !important;
  clip: rect(1px, 1px, 1px, 1px);
  pointer-events: none;
}
._3xxbHuF {
  max-width: 940px;
  margin: 0 auto;
}
.GN6nQol {
  padding: 18px 20px;
}
@media only screen and (min-width: 740px) {
  .GN6nQol {
    padding-left: 25px;
    padding-right: 25px;
  }
}
.GN6nQol._2hJYZLX {
  margin: 0 -20px;
}
@media only screen and (min-width: 740px) {
  .GN6nQol._2hJYZLX {
    margin: 0 -25px;
  }
}
.GN6nQol._1oH2W9O {
  padding-top: 0;
  padding-bottom: 0;
}
.GN6nQol.X0FctiP {
  color: #169400;
}
.GN6nQol.X0FctiP.spD5Qro {
  background-color: #169400;
  color: #fff;
}
.GN6nQol.X0FctiP._1-j2JXv {
  background-color: #f3f9f2;
}
.GN6nQol.llTEFEA {
  color: #9556b7;
}
.GN6nQol.llTEFEA.spD5Qro {
  background-color: #9556b7;
  color: #fff;
}
.GN6nQol.llTEFEA._1-j2JXv {
  background-color: #f9f6fb;
}
.GN6nQol._2pnoIks {
  color: #e60278;
}
.GN6nQol._2pnoIks.spD5Qro,
.GN6nQol._2pnoIks._1-j2JXv {
  background-color: #fff;
  color: #e60278;
}
.GN6nQol._1cG8CyK {
  color: #404040;
}
.GN6nQol._1cG8CyK.spD5Qro,
.GN6nQol._1cG8CyK._1-j2JXv {
  background-color: #fbfbfb;
  color: #404040;
}
.soHOhTq {
  padding-top: 30px;
  padding-bottom: 30px;
}
@media only screen and (min-width: 740px) {
  .soHOhTq {
    padding-top: 42px;
    padding-bottom: 36px;
  }
}
._31viNaz {
  padding-bottom: 12px;
  display: block;
}
._31viNaz ._31viNaz {
  position: relative;
  outline: 2px solid red;
}
._31viNaz ._31viNaz:after {
  content: "ERROR: Multiple <Text> components nested";
  background: red;
  font-weight: 700;
  color: #fff;
  position: absolute;
  top: 0;
  right: 0;
  padding: 0 20px;
}
._2R8GxMH {
  padding-bottom: 6px;
}
@media only screen and (max-width: 739px) {
  ._2R8GxMH:not(._3CnSGKb) {
    font-size: 16px;
    line-height: 24px;
    font-weight: 400;
  }
}
@media only screen and (min-width: 740px) {
  ._2R8GxMH:not(._3CnSGKb) {
    font-size: 16px;
    line-height: 24px;
    font-weight: 400;
  }
}
@media only screen and (max-width: 739px) {
  ._2R8GxMH._3CnSGKb {
    font-size: 16px;
    line-height: 24px;
    -webkit-transform: translateY(0.41em);
    transform: translateY(0.41em);
    font-weight: 400;
  }
}
@media only screen and (min-width: 740px) {
  ._2R8GxMH._3CnSGKb {
    font-size: 16px;
    line-height: 24px;
    -webkit-transform: translateY(0.41em);
    transform: translateY(0.41em);
    font-weight: 400;
  }
}
.NoROQmc {
  padding-bottom: 6px;
}
@media only screen and (max-width: 739px) {
  .NoROQmc:not(._3CnSGKb) {
    font-size: 14px;
    line-height: 18px;
    font-weight: 400;
  }
}
@media only screen and (min-width: 740px) {
  .NoROQmc:not(._3CnSGKb) {
    font-size: 14px;
    line-height: 18px;
    font-weight: 400;
  }
}
@media only screen and (max-width: 739px) {
  .NoROQmc._3CnSGKb {
    font-size: 14px;
    line-height: 18px;
    -webkit-transform: translateY(0.30285714em);
    transform: translateY(0.30285714em);
    font-weight: 400;
  }
}
@media only screen and (min-width: 740px) {
  .NoROQmc._3CnSGKb {
    font-size: 14px;
    line-height: 18px;
    -webkit-transform: translateY(0.30285714em);
    transform: translateY(0.30285714em);
    font-weight: 400;
  }
}
._1gBZb9f {
  padding-bottom: 6px;
}
@media only screen and (max-width: 739px) {
  ._1gBZb9f:not(._3CnSGKb) {
    font-size: 18px;
    line-height: 30px;
    font-weight: 400;
  }
}
@media only screen and (min-width: 740px) {
  ._1gBZb9f:not(._3CnSGKb) {
    font-size: 18px;
    line-height: 30px;
    font-weight: 400;
  }
}
@media only screen and (max-width: 739px) {
  ._1gBZb9f._3CnSGKb {
    font-size: 18px;
    line-height: 30px;
    -webkit-transform: translateY(0.49333333em);
    transform: translateY(0.49333333em);
    font-weight: 400;
  }
}
@media only screen and (min-width: 740px) {
  ._1gBZb9f._3CnSGKb {
    font-size: 18px;
    line-height: 30px;
    -webkit-transform: translateY(0.49333333em);
    transform: translateY(0.49333333em);
    font-weight: 400;
  }
}
@media only screen and (max-width: 739px) {
  ._3X2sUcS:not(._3CnSGKb) {
    font-size: 18px;
    line-height: 24px;
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._3X2sUcS:not(._3CnSGKb) {
    font-size: 18px;
    line-height: 24px;
    font-weight: 700;
  }
}
@media only screen and (max-width: 739px) {
  ._3X2sUcS._3CnSGKb {
    font-size: 18px;
    line-height: 24px;
    -webkit-transform: translateY(0.32666667em);
    transform: translateY(0.32666667em);
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._3X2sUcS._3CnSGKb {
    font-size: 18px;
    line-height: 24px;
    -webkit-transform: translateY(0.32666667em);
    transform: translateY(0.32666667em);
    font-weight: 700;
  }
}
@media only screen and (max-width: 739px) {
  ._1iMwyee:not(._3CnSGKb) {
    font-size: 21px;
    line-height: 30px;
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._1iMwyee:not(._3CnSGKb) {
    font-size: 28px;
    line-height: 36px;
    font-weight: 700;
  }
}
@media only screen and (max-width: 739px) {
  ._1iMwyee._3CnSGKb {
    font-size: 21px;
    line-height: 30px;
    -webkit-transform: translateY(0.37428571em);
    transform: translateY(0.37428571em);
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._1iMwyee._3CnSGKb {
    font-size: 28px;
    line-height: 36px;
    -webkit-transform: translateY(0.30285714em);
    transform: translateY(0.30285714em);
    font-weight: 700;
  }
}
@media only screen and (max-width: 739px) {
  ._12x7M_H:not(._3CnSGKb) {
    font-size: 21px;
    line-height: 30px;
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._12x7M_H:not(._3CnSGKb) {
    font-size: 22px;
    line-height: 30px;
    font-weight: 700;
  }
}
@media only screen and (max-width: 739px) {
  ._12x7M_H._3CnSGKb {
    font-size: 21px;
    line-height: 30px;
    -webkit-transform: translateY(0.37428571em);
    transform: translateY(0.37428571em);
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._12x7M_H._3CnSGKb {
    font-size: 22px;
    line-height: 30px;
    -webkit-transform: translateY(0.34181818em);
    transform: translateY(0.34181818em);
    font-weight: 700;
  }
}
@media only screen and (max-width: 739px) {
  ._2gZSEA6:not(._3CnSGKb) {
    font-size: 21px;
    line-height: 30px;
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._2gZSEA6:not(._3CnSGKb) {
    font-size: 42px;
    line-height: 48px;
    font-weight: 400;
  }
}
@media only screen and (max-width: 739px) {
  ._2gZSEA6._3CnSGKb {
    font-size: 21px;
    line-height: 30px;
    -webkit-transform: translateY(0.37428571em);
    transform: translateY(0.37428571em);
    font-weight: 700;
  }
}
@media only screen and (min-width: 740px) {
  ._2gZSEA6._3CnSGKb {
    font-size: 42px;
    line-height: 48px;
    -webkit-transform: translateY(0.23142857em);
    transform: translateY(0.23142857em);
    font-weight: 400;
  }
}
._3sPkVJ8 {
  font-weight: 300;
}
._3ScsQxm {
  padding-bottom: 0;
}
.JSGiS7B {
  display: list-item;
  list-style-type: disc;
  margin-left: 30px;
}
._24Qjjfm {
  color: #169400;
}
._1PaeMXM {
  color: #e60278;
}
._2fm-_V2 {
  color: #747474;
}
.sc2jJmq {
  border-radius: 2px;
  display: inline-block;
  margin: -2px;
  padding: 2px;
  background-color: rgba(30, 144, 255, 0.12);
}
._23NNZdd {
  color: #404040;
}
._3xZ2xhz {
  background-color: rgba(230, 2, 120, 0.12);
}
._29oJaV- {
  color: #9556b7;
}
._1lTs4d6 {
  font-weight: 700;
}
._33Weyvq {
  font-weight: 400;
}
._2f9d9Gi {
  -webkit-animation-duration: 0.4s;
  animation-duration: 0.4s;
  -webkit-animation-name: l73EdLW;
  animation-name: l73EdLW;
}
@-webkit-keyframes l73EdLW {
  0% {
    -webkit-transform: scale(0.5);
    transform: scale(0.5);
  }
  65% {
    -webkit-transform: scale(1.5);
    transform: scale(1.5);
  }
  85% {
    -webkit-transform: scale(0.8);
    transform: scale(0.8);
  }
  to {
    -webkit-transform: scale(1);
    transform: scale(1);
  }
}
@keyframes l73EdLW {
  0% {
    -webkit-transform: scale(0.5);
    transform: scale(0.5);
  }
  65% {
    -webkit-transform: scale(1.5);
    transform: scale(1.5);
  }
  85% {
    -webkit-transform: scale(0.8);
    transform: scale(0.8);
  }
  to {
    -webkit-transform: scale(1);
    transform: scale(1);
  }
}
._3q2EWBs {
  display: inline-block;
}
.URJCV7U {
  fill: currentColor;
  width: 16px;
  height: 16px;
}
._2MMt39x {
  width: 29.4px;
  height: 29.4px;
}
@media only screen and (max-width: 739px) {
  ._2MMt39x {
    width: 14.7px;
    height: 14.7px;
  }
}
._3Q7FqJ4 {
  width: 19.6px;
  height: 19.6px;
}
@media only screen and (max-width: 739px) {
  ._3Q7FqJ4 {
    width: 14.7px;
    height: 14.7px;
  }
}
._3Lp4JO5 {
  width: 12.6px;
  height: 12.6px;
}
@media only screen and (max-width: 739px) {
  ._3Lp4JO5 {
    width: 12.6px;
    height: 12.6px;
  }
}
._2mhPJKN {
  width: 15.4px;
  height: 15.4px;
}
@media only screen and (max-width: 739px) {
  ._2mhPJKN {
    width: 14.7px;
    height: 14.7px;
  }
}
._1my9l-4 {
  width: 12.6px;
  height: 12.6px;
}
@media only screen and (max-width: 739px) {
  ._1my9l-4 {
    width: 12.6px;
    height: 12.6px;
  }
}
._2aC_WIb {
  width: 11.2px;
  height: 11.2px;
}
@media only screen and (max-width: 739px) {
  ._2aC_WIb {
    width: 11.2px;
    height: 11.2px;
  }
}
._2RscG1n {
  width: 9.8px;
  height: 9.8px;
}
@media only screen and (max-width: 739px) {
  ._2RscG1n {
    width: 9.8px;
    height: 9.8px;
  }
}
.kisKgQ5 {
  width: 12.6px;
  height: 12.6px;
}
@media only screen and (max-width: 739px) {
  .kisKgQ5 {
    width: 12.6px;
    height: 12.6px;
  }
}
,
  "vendors~main-7efe22f8be0a6ad2b461.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;
